src/Autocomplete/src/AutocompleteResultsExecutor.php

```diff
@@ -46,14 +50,57 @@ public function fetchResults(EntityAutocompleterInterface $autocompleter, string
 
         $entities = $queryBuilder->getQuery()->execute();
 
-        $results = [];
+        $options = [];
+        $optgroupLabels = [];
+
+        if (null !== $groupBy = $autocompleter->getFormOption('group_by')) {
+            if (\is_string($groupBy)) {
+                $groupBy = new PropertyPath($groupBy);
+            }
+
+            if ($groupBy instanceof PropertyPathInterface) {
+                $accessor = PropertyAccess::createPropertyAccessor();
+                $groupBy = function ($choice) use ($accessor, $groupBy) {
+                    try {
+                        return $accessor->getValue($choice, $groupBy);
+                    } catch (UnexpectedTypeException) {
+                        return null;
+                    }
+                };
+            }
+
+            if (\is_callable($groupBy)) {
```

Comment:
And what happens if `$groupBy` is not a callable? Should that trigger an error?

Reply:
I assumed that if ever it is not callable then it falls back to the simple case without group_by but, indeed, we can trigger an error, which do you prefer?